src/NextGen/Gateways/PayPalCommerce/payPalCommerceGateway.tsx

```diff
@@ -0,0 +1,327 @@
+    }
+
+    const createSubscriptionHandler = async (data, actions) => {
+        // eslint-disable-next-line
+        const response = await fetch( `${ payPalDonationsSettings.ajaxurl }?action=give_paypal_commerce_create_plan_id`, {
+            method: 'POST',
+            body: getFormData(),
+        } );
+
+        const responseJson = await response.json();
+
+        if ( ! responseJson.success ) {
+            throw responseJson.data.error;
+        }
+
+        payPalSubscriptionId = responseJson.data.id;
+
+        return actions.subscription.create( { plan_id: payPalSubscriptionId  } );
+    }
+
+    const Divider = ({label, style = {}}) => {
+
+        const styles = {
+            container: {
+                fontSize: '16px',
+                fontStyle: 'italic',
+                display: 'flex',
+                justifyContent: 'center',
+                alignItems: 'center',
+                ...style,
+            },
+            dashedLine: {
+                border: '1px solid #d4d4d4',
+                flexGrow: 1,
+            },
+            label: {
+                padding: '0 6px',
+                fontSize: '14px',
+                color: '#8d8e8e',
+            },
+        }
+
+        return (
+            <div className="separator-with-text" style={styles.container}>
+                <div className="dashed-line" style={styles.dashedLine} />
+                <div className="label" style={styles.label}>
+                    {label}
+                </div>
+                <div className="dashed-line" style={styles.dashedLine} />
+            </div>
+        )
+    }
+
+    const HoistHostedFieldContext = () => {
+        hostedField = usePayPalHostedFields();
+        return <></>
+    }
+
+    const FormFieldsProvider = ({children}) => {
+        const {useWatch} = window.givewp.form.hooks;
+        amount = useWatch({name: 'amount'});
+        firstName = useWatch({name: 'firstName'});
+        lastName = useWatch({name: 'lastName'});
+        email = useWatch({name: 'email'});
+
+        return children;
+    }
+
+    const SmartButtonsContainer = () => {
+
+        const {useWatch, useFormState} = window.givewp.form.hooks;
+        const currency = useWatch({name: 'currency'});
+        const donationType = useWatch({name: 'donationType'});
+
+        const {
+            isSubmitting
+        } = useFormState();
+
+        const [{ options }, dispatch] = usePayPalScriptReducer();
+
+        useEffect(() => {
+            dispatch({
+                type: "resetOptions",
+                value: {
+                    ...options,
+                    currency: currency,
+                    vault: donationType === 'subscription',
+                    intent: donationType === 'subscription' ? 'subscription' : 'capture',
+                },
+            });
+        }, [currency, donationType]);
+
+        return (
+            <PayPalButtons
+                disabled={isSubmitting}
```

Comment:
I ended up making this change to the form's donate button in the epic recurring PR recently because I noticed a delay between these two - it makes sure the button _stays_ disabled while redirecting to the confirmation page.

```suggestion
                disabled={isSubmitting || isSubmitSuccessful}
```

Reply:
Oh, yeah - good idea!